hnswlib/space_l2.h

```diff
@@ -19,7 +19,38 @@ namespace hnswlib {
         return (res);
     }
 
-#if defined(USE_AVX)
+#if defined(USE_AVX512)
+
+    // Favor using AVX512 if available.
+    static float
+    L2SqrSIMD16Ext(float *pVect1, float *pVect2, size_t qty) {
+        float PORTABLE_ALIGN32 TmpRes[16];
+        size_t qty16 = qty >> 4;
+
+        const float *pEnd1 = pVect1 + (qty16 << 4);
+
+        __m512 diff, v1, v2;
+        __m512 sum = _mm512_set1_ps(0);
+
+        while (pVect1 < pEnd1) {
+            v1 = _mm512_loadu_ps(pVect1);
+            pVect1 += 16;
+            v2 = _mm512_loadu_ps(pVect2);
+            pVect2 += 16;
+            diff = _mm512_sub_ps(v1, v2);
+            // sum = _mm512_fmadd_ps(diff, diff, sum);
```

Comment:
I wonder why the line is commented? Does it perform slower?

Reply:
Yes - I tested a fused multiply-add and it was slower for the AVX2 code. Was the same for the AVX512 code, so I decided to not use it.